sqlglot/dialects/starrocks.py

```diff
@@ -70,6 +79,35 @@ def _parse_unnest(self, with_alias: bool = True) -> t.Optional[exp.Unnest]:
 
             return unnest
 
+        def _parse_interval_or_number(self) -> t.Optional[exp.Expression]:
+            return self._parse_interval(keep_number=True) or self._parse_number()
+
+        def _parse_partitioning_granularity_dynamic(self) -> exp.PartitionByRangePropertyDynamic:
+            self._match_text_seq("START")
+            start = self._parse_wrapped(self._parse_string)
+            self._match_text_seq("END")
+            end = self._parse_wrapped(self._parse_string)
+            self._match_text_seq("EVERY")
+            every = self._parse_wrapped(self._parse_interval_or_number)
+            return self.expression(
+                exp.PartitionByRangePropertyDynamic, start=start, end=end, every=every
+            )
+
+        def _parse_partition_by_opt_range(
+            self,
+        ) -> exp.PartitionedByProperty | exp.PartitionByRangeProperty:
+            if self._match_text_seq("RANGE"):
+                partition_expressions = self._parse_wrapped_id_vars()
+                create_expressions = self._parse_wrapped_csv(
+                    self._parse_partitioning_granularity_dynamic
+                )
+                return self.expression(
+                    exp.PartitionByRangeProperty,
+                    partition_expressions=partition_expressions,
+                    create_expressions=create_expressions,
+                )
+            return super()._parse_partitioned_by()
```

Comment:
Just double-checking: there's also this in starrocks' docs:

```
PARTITION <partition1_name> VALUES LESS THAN ("<upper_bound_for_partitioning_column1>" [ , "<upper_bound_for_partitioning_column2>", ... ] )
```

From what I understand, your changes only add support for the `(START...END...EVERY...)` syntax.

1. What happens today if the user tries to parse the alternative syntax?
2. Same question as (1), but given the changes in your PR?

I wanna ensure there are no regressions.

Reply:
I did not support this as I don't have the need to use it, other people that need it can add it later, later they can just add another expression for this, since it is totally different.

Yup, it only supports the dynamic partitioning range method `(START...END...EVERY...)` syntax like you mentioned.

Reply:
This makes sense, but there's some nuance here. Some unsupported DDLs are still "parsed" today, meaning that we produce a `Command` node and store the input as raw text in it. The motivation behind this is to avoid crashing in cases where transpilation / parsing is not required for certain statements.

Since you introduced new logic for this property, I wanted to ensure that we don't break these `Command` statements accidentally by entering this `if self._match_text_seq("RANGE")` branch.

Basically let's test what happened before vs after. If the `VALUES LESS THAN` clause was not supported before, it's fine. If it was parsed into a `Command`, we need to make sure that if the parser fails, we'll fallback to a command as well.

Does this make sense?

Reply:
I think I will ignore this for now, don't want to spend too much time on this.